### PR TITLE
Move the CI from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by oliverklee**
Travis CI has dropped free support for open source projects.

To continue having CI builds, we now use GitHub Actions for this.

### CODE REVIEW

**Comment by oliverklee** [comment]
Coverage support will come in a later PR.

**Comment by Danon** [comment]
@oliverklee 
> Travis CI has dropped free support for open source projects.

Do you have a source for that, that I can read?

**Comment by Danon** [comment]
@oliverklee 

> Coverage support will come in a later PR.

Can you prepare the PR for the coverage as well? I will merge them in the right order

**Comment by oliverklee** [comment]
> Do you have a source for that, that I can read?

Yes, here: https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

**Comment by oliverklee** [comment]
> Can you prepare the PR for the coverage as well? I will merge them in the right order

Probably not, as the coverage PR will need the changes from this PR.

**Comment by Danon** [comment]
> > Can you prepare the PR for the coverage as well? I will merge them in the right order
> 
> Probably not, as the coverage PR will need the changes from this PR.

Base your coverage branch on this branch :)

**Comment by Danon** [comment]
> > Do you have a source for that, that I can read?
> 
> Yes, here: https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

Can you give me some source from travis CI website?

**Comment by oliverklee** [comment]
https://travis-ci.org/

> Please be aware travis-ci.org will be shutting down in several weeks, with all accounts migrating to travis-ci.com. Please stay tuned here for more information. 

https://travis-ci.com/account/plan

The free plan provides 10000 build credits. Once they are used up (which happened to me with my open source projects), Travis does not build the projects anymore.

**Comment by Danon** [comment]
> The free plan provides 10000 build credits. Once they are used up (which happened to me with my open source projects), Travis does not build the projects anymore.

Do the 10000 credits refresh, say each month or each year? Or do I only ever get 10000 credits? And when they're gone I've got nothing but to buy more?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: '3 3 * * 1'
```
Comment (Danon) [inline .github/workflows/ci.yml:7]: Could you remove the `cron` part, please? I'd hate to have this running by it's own.

**Comment by oliverklee** [comment]
> Do the 10000 credits refresh, say each month or each year?

They don't refresh for the free plan AFAIK.

> Or do I only ever get 10000 credits? And when they're gone I've got nothing but to buy more?

Correct.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (oliverklee) [reply .github/workflows/ci.yml]: Done and repushed.

(I find the cron jobs useful to detect breakage that is related to dependencies or the changes CI environment. AS this project does not have any third-party dependencies, this seems to be less important here.)

**Comment by Danon** [comment]
@oliverklee How do I do the same with https://github.com/T-Regx/T-Regx ?

**Comment by oliverklee** [comment]
@Danon It's mostly doing the same as in #6 and #7:

1. copy the `.github/` folder
2. adapt the PHP versions in the `ci.yml` file as needed
3. remove `.travis.yml`
4. adapt the `.gitattributes` laccordingly
5. update the "CI status" button in the README like in this project

If you're stuck at some point, you can ping me.

**Pull request merged**